### PR TITLE
Build client on Windows in a temporary directory

### PR DESCRIPTION
**Pull request by johnstep**
Signed-off-by: John Stephens <johnstep@docker.com>

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Sooooo much better. I think it could do with a couple of tweaks to not be so verbose (using `--quiet`) and catch any git errors. Something like this:

```PowerShell
            # Temporarily override GOPATH, then clone, checkout, and build.
            $saveGOPATH = $env:GOPATH
            Try {
                $env:GOPATH = $tempLocation
                $dockerCliRoot = "$env:GOPATH\src\$($dockerCliRepo.Split("/", 3)[2])"
                Write-Host "INFO: Cloning client repository from $dockerCliRepo..."
                git clone --quiet $dockerCliRepo $dockerCliRoot
                if ($LASTEXITCODE -ne 0) { Throw "Failed to clone client repository" }
                Write-Host "INFO: Checking out client commit $dockerCliCommit..."
                git -C $dockerCliRoot -c advice.detachedHead=false checkout --quiet $dockerCliCommit
                if ($LASTEXITCODE -ne 0) { Throw "Failed to checkout client commit $dockerCliCommit" }
                Write-Host "INFO: Building client..."
                go build -o "$root\bundles\docker.exe" "github.com/docker/cli/cmd/docker"
                if ($LASTEXITCODE -ne 0) { Throw "Failed to compile client" }
            }
            Catch [Exception] {
                Throw $_
            }
            Finally {
                # Always restore GOPATH and remove the temporary directory.
                $env:GOPATH = $saveGOPATH
                Remove-Item -Force -Recurse $tempLocation
            }
```

File: hack/make.ps1 [hunk 1]
```diff
@@ -397,16 +397,36 @@ Try {
...
+            $tempLocation = "$env:TEMP\$(New-Guid)"
+            New-Item -ItemType Directory $tempLocation | Out-Null
+
+            # Temporarily override GOPATH, then clone, checkout, and build.
+            $saveGOPATH = $env:GOPATH
+            Try {
+                $env:GOPATH = $tempLocation
+                $dockerCliRoot = "$env:GOPATH\src\$($dockerCliRepo.Split("/", 3)[2])"
+                Write-Host "INFO: Cloning client repository..."
+                git clone $dockerCliRepo $dockerCliRoot
```
Comment (jhowardmsft) [inline hack/make.ps1]: Errors here need to be captured too.

File: hack/make.ps1 [hunk 2]
```diff
@@ -397,16 +397,36 @@ Try {
...
+                git -C $dockerCliRoot -c advice.detachedHead=false checkout $dockerCliCommit
```
Comment (jhowardmsft) [inline hack/make.ps1]: Same here

**Review by lowenna — CHANGES_REQUESTED** [review]
See larger comment, but looks pretty good. Just a few nits need addressing.

**Review by lowenna — COMMENTED** [review]
LGTM 😸

**Review by lowenna — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**